File: network-topology-frontend/src/types/topology.ts
export interface GeoData {
  country: string | null;
  city: string | null;
  lat: number | null;
  lng: number | null;
}

export interface ASNData {
  number: number | null;
  organization: string | null;
}

export interface HopResponse {
  hop: number;
  ip: string;
  rtt_ms: number | null;
  status: 'success' | 'timeout' | 'error';
  geo: GeoData;
  asn: ASNData;
  didatic_explanation: string;
}

export interface WebNode {
  url: string;
  domain: string;
  category: 'internal' | 'subdomain' | 'external';
  resource_type: 'document' | 'image' | 'script' | 'stylesheet';
  didatic_explanation: string;
}

export interface DNSRecords {
  A: string[];
  NS: string[];
  MX: string[];
  TXT: string[];
}

export interface TLSData {
  issuer: string;
  subject: string;
  version: string;
  cipher: string;
  valid_until: string;
}

File: network-topology-backend/app/schemas/topology.py
from pydantic import BaseModel, Field
from typing import Optional

class GeoData(BaseModel):
    country: Optional[str] = Field(None, description="Codigo do pais (ex: BR)")
    city: Optional[str] = Field(None, description="Nome da cidade")
    lat: Optional[float] = Field(None, description="Latitude")
    lng: Optional[float] = Field(None, description="Longitude")

class ASNData(BaseModel):
    number: Optional[int] = Field(None, description="Numero do Sistema Autonomo (ASN)")
    organization: Optional[str] = Field(None, description="Nome da organizacao dona do bloco IP")

class HopResponse(BaseModel):
    hop: int = Field(..., description="Numero do salto atual")
    ip: str = Field(..., description="Endereco IP do roteador intermediario")
    rtt_ms: Optional[float] = Field(None, description="Round Trip Time em milissegundos")
    status: str = Field(..., description="Status do salto: success, timeout, ou error")
    geo: GeoData
    asn: ASNData
    didatic_explanation: str = Field(..., description="Explicacao didatica sobre o evento ocorrido neste salto")